### PR TITLE
Fixed generation of output directory

### LINKED ISSUE #106: Bug when generating ${PATH}_ddr directory
The problem is the same as in https://github.com/objectionary/aoi/issues/43. It turned out that https://github.com/objectionary/aoi/issues/43 related to a problem with `createTempDirectories` private function from `org.objectionary.ddr.launch`. Therefore, this problem needs to be fixed here.

### PR DESCRIPTION
**Pull request by c71n93**
Closes #106 and also https://github.com/objectionary/aoi/issues/43

### CODE REVIEW

**Comment by c71n93** [comment]
@yegor256 please take a look

**Comment by yegor256** [comment]
@c71n93 how do you know anything is fixed? I don't see any changes in unit tests. They all worked before, they all work now. What's the purpose of your changes?

**Comment by c71n93** [comment]
> @c71n93 how do you know anything is fixed? I don't see any changes in unit tests. They all worked before, they all work now. What's the purpose of your changes?

I understand. I will add unit tests that cover this behavior.

**Comment by c71n93** [comment]
@yegor256 I added tests, take a look, please

**Comment by yegor256** [comment]
@mximp please, review this one

File: src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ class InnerPropagator(
     private fun processDecorators() {
         val repetitions = 5
         // @todo #44:30min this solution is naive, optimize it (see commented out lines)
```
Comment (mximp) [inline src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt:67]: @c71n93 According to our convention TODOs are written as part of JavaDoc comment for the nearest method/class. In general intra-method comments are discouraged.

File: src/main/kotlin/org/objectionary/ddr/launch/Combiner.kt [hunk 2]
```diff
@@ -48,11 +48,11 @@ val documents: MutableMap<Document, String> = mutableMapOf()
  * Aggregates all steps of analysis
  *
  * @param path path to the directory to be analysed
- * @param dirName postfix of the resulting directory
+ * @param dirPostfix postfix of the resulting directory
  */
-fun launch(path: String, dirName: String = "ddr") {
+fun launch(path: String, dirPostfix: String = "ddr") {
```
Comment (mximp) [inline src/main/kotlin/org/objectionary/ddr/launch/Combiner.kt:53]: @c71n93 It's better to pick simple nouns for variables. If a variable has compound name it most probably has too wide visibility scope. In this case: can `dirPostfix` be renamed into just `postfix`?

Comment (c71n93) [reply src/main/kotlin/org/objectionary/ddr/launch/Combiner.kt]: @mximp I agree, the `postfix` name will be enough here.

File: src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt (hunk 1 above)
Comment (c71n93) [reply src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt]: @mximp fixed this

File: src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt [hunk 3]
```diff
@@ -61,10 +61,12 @@ class InnerPropagator(
...
     }
 
+    /**
+     * @todo #44:30min this solution is naive, optimize it (see commented out lines)
+     */
     @Suppress("MAGIC_NUMBER")
     private fun processDecorators() {
         val repetitions = 5
-        // @todo #44:30min this solution is naive, optimize it (see commented out lines)
         // while (decorators.containsValue(false)) {
```
Comment (mximp) [inline src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt:70]: @c71n93 do we really need this comment here?

Comment (c71n93) [reply src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt:70]: @mximp without this comment, it is not entirely clear what was meant in TODO. However, to be honest, it's still unclear with the comment. Can I fix this in the next PRs, when I need to figure out what this code does?

Comment (mximp) [reply src/main/kotlin/org/objectionary/ddr/graph/InnerPropagator.kt:70]: @c71n93 Just move the commented snippet along with TODO

**Review by mximp — APPROVED** [review]

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**